src/react/boosts/payments/BoostPayFormStripe.tsx

```diff
@@ -96,61 +230,189 @@ class BoostPayFormStripe extends React.Component<BoostPayFormStripeProps> {
+        <FullScreenModal open={this.state.isSuccessModalOpen}>
+          <BoostModalContain>
+            <PaymentSuccessCardModalText
+              newsroomName={this.props.newsroomName}
+              usdToSpend={this.props.usdToSpend}
+              handlePaymentSuccess={this.props.handlePaymentSuccess}
+            />
+          </BoostModalContain>
+        </FullScreenModal>
+      </form>
     );
   }
 
+  private hideModal = () => {
+    this.setState({ isErrorModalOpen: false, supportDisabled: false });
+  };
+
+  private handleOnChange = (event: any) => {
+    const state = event.target.id;
+    const value = event.target.value;
+
+    switch (state) {
+      case "email":
+        this.setState({ email: value });
+        break;
+      case "name":
+        this.setState({ name: value });
+        break;
+      case "country":
+        this.setState({ country: value });
+        break;
+      case "zip":
+        this.setState({ postalCode: value });
+        break;
+      default:
+        break;
+    }
+  };
+
+  private isValidPostalCode = () => {
+    const postalCode = this.state.postalCode.toString().trim();
+    const usa = /^[0-9]{5}(?:-[0-9]{4})?$/;
+    const can = /^[ABCEGHJKLMNPRSTVXY]\d[ -]?\d[A-Za-z]\d$/;
+    const gbr = /^[A-Z]{1,2}[0-9]{1,2} ?[0-9][A-Z]{2}$/i;
+
+    switch (this.state.country) {
+      case "USA":
+        return usa.test(postalCode);
+      case "CAN":
+        return can.test(postalCode);
+      case "GBR":
+        return gbr.test(postalCode);
+      default:
+        return true;
+    }
+  };
+
   private async handleSubmit(): Promise<void> {
-    if (this.props.stripe) {
-      try {
-        const token = await this.props.stripe.createToken();
-        await this.props.savePayment({
-          variables: {
-            postID: this.props.boostId,
-            input: { paymentToken: token, amount: this.props.usdToSpend, currencyCode: "usd" },
-          },
-        });
-      } catch (err) {
-        console.error(err);
+    const validEmail = isValidEmail(this.state.email);
+    const validName = this.state.name !== "";
+    const validCountry = this.state.country !== "";
+    const validPostalCode = this.isValidPostalCode();
+
+    if (!validEmail || !validName || !validCountry || !validPostalCode) {
+      this.setState({ validEmail, validName, validCountry, validPostalCode });
+    } else {
+      this.setState({ supportDisabled: true });
+      if (this.props.stripe) {
+        try {
+          const token = await this.props.stripe.createToken({
+            name: this.state.name,
+            address_country: this.state.country,
+            address_zip: this.state.postalCode,
+          });
+          await this.props
+            .savePayment({
+              variables: {
+                postID: this.props.boostId,
+                input: { paymentToken: token, amount: this.props.usdToSpend, currencyCode: "usd" },
+              },
+            })
+            .then(() => {
```

Comment:
Feels a bit weird to use both async/await style and promise callbacks, I'd stick with just one (and we tend to only use async/await unless in a function that can't be async for some reason like react component render).

`await` means it doesn't get to the next line of code until/unless the promise resolves successfully, even if you're not consuming the return value from the promise, so this is the same without the callback:

```
await this.props.savePayment(...);
this.setState(...);
```